apply_kernel.json

```
@@ -17,7 +17,7 @@
         },
         {
             "name": "kernel",
-            "description": "A two-dimensional weighted kernel. Each dimension of the kernel must have an uneven number of elements, otherwise the process throws a `KernelDimensionsUneven` error.",
+            "description": "Kernel as a two-dimensional array of weights. The inner level of the nested array corresponds to the `x` dimension and the outer level corresponds to the `y` dimension. Each dimension of the kernel must have an uneven number of elements, otherwise the process throws a `KernelDimensionsUneven` error.",
```

Referring to the x and y dimension feels a bit unclear. In the metadata it's called x and y axis. If we way dimension, it sounds a bit like the dimension needs to be names as such, but that's not the case although that's the suggested default.

```suggestion
            "description": "Kernel as a two-dimensional array of weights. The inner level of the nested array corresponds to the `x` axis and the outer level corresponds to the `y` axis. Each dimension of the kernel must have an uneven number of elements, otherwise the process throws a `KernelDimensionsUneven` error.",
```

---

what about 

> horizontal spatial dimension (commonly called `x`) and vertical spatial dimension (commonly called `y`)

?

---

Hm, y is not really vertical? x and y are both horiztonal. Throughout the API z is the vertical axis.

---

hmm indeed, I'm too stuck in my 2d world :)

is it common to speak of *first* and *second* spatial dimension? (so without using "horizontal" because that could introduce confusion)

---

Not sure. I guess the long form would be:

> The inner level of the nested array corresponds to the dimension with the axis set to `x` and the outer level corresponds to the dimension with the axis set to `y`.

That is at least very close to your metadata, where the horizontal dimensions have a property "axis" set to "x" or "y".

---

To not make it too confusing I went with your first suggestion 

> The inner level of the nested array aligns with the `x` axis ...

(force pushed amended commit)